rmk/src/hid.rs

```diff
@@ -40,6 +42,46 @@ pub(crate) trait HidWriterWrapper: ConnectionTypeWrapper {
 }
 
 pub(crate) trait HidReaderWriterWrapper: HidReaderWrapper + HidWriterWrapper {}
+impl<T: HidReaderWrapper + HidWriterWrapper> HidReaderWriterWrapper for T {}
+
+pub(crate) struct ReaderWriter<T1, T2>(pub T1, pub T2);
```

Comment:
What about doing:

```rust
pub(crate) struct EspBleReaderWriter<'ch, M: RawMutex, T: Sized, const N: usize, W: HidWriterWrapper> {
    receiver: Receiver<'ch, M, T, N>,
    hid_writer: W,
}

impl<'ch, M: RawMutex, T: Sized, const N: usize, W: HidWriterWrapper> ConnectionTypeWrapper
    for EspBleReaderWriter<'ch, M, T, N, W>
{
    fn get_conn_type(&self) -> ConnectionType {
        ConnectionType::Ble
    }
}

impl<'ch, M: RawMutex, T: Sized, const N: usize, W: HidWriterWrapper> HidReaderWrapper
    for EspBleReaderWriter<'ch, M, T, N, W>
{
    async fn read(&mut self, buf: &mut [u8]) -> Result<usize, HidError> {
        let v = self.receiver.receive().await;
        buf.copy_from_slice(as_bytes(&v));
        Ok(as_bytes(&v).len())
    }
}

impl<'ch, M: RawMutex, T: Sized, const N: usize, W: HidWriterWrapper> HidWriterWrapper
    for EspBleReaderWriter<'ch, M, T, N, W>
{
    async fn write_serialize<IR: AsInputReport>(&mut self, r: &IR) -> Result<(), HidError> {
        self.hid_writer.write_serialize(r).await
    }

    async fn write(&mut self, report: &[u8]) -> Result<(), HidError> {
        self.hid_writer.write(report).await
    }
}
```
And put it to `esp` mod?

Then you can use it as:

```rust
let mut via_rw = EspBleReaderWriter::new(via_output.receiver(), ble_server.input_vial);
```

What do you think?